### PR TITLE
Fixing the display of deleted files in pull requests

### PR DESCRIPTION
**Pull request by StanleyGoldman**
Fixing the issue mentioned in #646

Image

### CODE REVIEW

**Comment by grokys** [comment]
Thanks @StanleyGoldman, only question is if we want to retain the style used elsewhere in Team Explorer where red is not used (which is what the current style is based on). If we decide we want to use a different style, I will merge this.

File: src/GitHub.VisualStudio/UI/Views/PullRequestDetailView.xaml [hunk 1]
```diff
@@ -448,7 +448,8 @@
                                                 <TextBlock.Style>
                                                     <Style TargetType="TextBlock">
                                                         <Style.Triggers>
-                                                            <DataTrigger Binding="{Binding ChangeType}" Value="Removed">
+                                                            <DataTrigger Binding="{Binding Status}" Value="Removed">
+                                                                <Setter Property="Foreground" Value="Red"/>
```
Comment (shana) [inline src/GitHub.VisualStudio/UI/Views/PullRequestDetailView.xaml]: Colors shouldn't be hardcoded, as the UI needs to adapt to theme changes. We have different files defining the colors for each theme under the `Styles` folder, any control that needs theme-aware colors should reference resources from there.

Comment (StanleyGoldman) [inline src/GitHub.VisualStudio/UI/Views/PullRequestDetailView.xaml]: Fixed up.

**Comment by StanleyGoldman** [comment]
@donokuda What kind of grey should be used instead?

**Comment by donokuda** [comment]
Hey @StanleyGoldman! I pushed 7063bfc to your branch (success! 🎉) to use the `VsColor.GrayText` brush for a more subdued gray. I chose VisualStudio's gray so we'll have the right color for each theme out of the box.

image

The only thing I couldn't figure out is updating the Octicon's `Foreground` to be the same gray when the file is deleted.

**Comment by shana** [comment]
> The only thing I couldn't figure out is updating the Octicon's Foreground to be the same gray when the file is deleted.

Oh, probably the same way as we're doing in the name, with an inline style with triggers to switch the background color in https://github.com/StanleyGoldman/VisualStudio/blob/7063bfc8a4f1837d686c52aeba3f2375f9a78fd6/src/GitHub.VisualStudio/UI/Views/PullRequestDetailView.xaml#L446

**Comment by shana** [comment]
This looks great! Perhaps we should add the icon color change as well in this PR and have it all nicely wrapped up, what do you think @StanleyGoldman?

**Comment by StanleyGoldman** [comment]
Yep, let's wrap this one up.

File: src/GitHub.VisualStudio/UI/Views/PullRequestDetailView.xaml [hunk 2]
```diff
@@ -443,7 +443,18 @@
...
                                                     ContextMenu="{StaticResource FileContextMenu}"
                                                     Tag="{Binding DataContext, RelativeSource={RelativeSource Mode=FindAncestor,AncestorType={x:Type te:SectionControl}}}">
-                                            <ui:OcticonImage Icon="file_code" Foreground="{DynamicResource GitHubVsWindowText}"/>
+                                            <ui:OcticonImage Icon="file_code" Foreground="{DynamicResource GitHubVsWindowText}">
+                                                <ui:OcticonImage.Style>
+                                                    <Style TargetType="ui:OcticonImage" BasedOn="{StaticResource OcticonImage}">
+                                                        <Style.Triggers>
+                                                            <DataTrigger Binding="{Binding Status}" Value="Removed">
+                                                                <Setter Property="Foreground" Value="{DynamicResource GitHubDeletedFileIconBrush}"/>
+                                                                <!--<Setter Property="Foreground" Value="Red"/>-->
```
Comment (StanleyGoldman) [inline src/GitHub.VisualStudio/UI/Views/PullRequestDetailView.xaml]: I'm having a bit of trouble with the binding here. I know I have the trigger right, as I'm able to see the effect of the value `Red` in the Visual Studio designer, but at runtime neither the DynamicResource binding nor the value `Red` seem to have an effect.

**Comment by StanleyGoldman** [comment]
@donokuda I put the grey on the icon, but I don't know if it stands out well enough. Feel free to change the color as you see fit.

image

**Comment by StanleyGoldman** [comment]
Yea, i see that point to the light grey icon doesn't really stand out with all that ~~strikeout~~ text right next to it.
I think it's good now though, the washed out icon has already grown on me. 😄

**Comment by donokuda** [comment]
@StanleyGoldman Awesome! Going to merge this now. Thanks for this! 🙇

**Pull request merged**